testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -173,6 +176,17 @@ protected boolean hasBehavior(TestingConnectorBehavior connectorBehavior)
         return connectorBehavior.hasBehaviorByDefault(this::hasBehavior);
     }
 
+    protected boolean hasCapability(Session session, ConnectorCapabilities capability)
+    {
+        CatalogName catalogName = session.getCatalog()
+                .map(CatalogName::new)
+                .orElseThrow();
+
+        getQueryRunner().getMetadata().getCatalogHandle(session, catalogName.getCatalogName());
```

Comment:
Is it needed?

Reply:
Yes, it is needed. It is a trick to register catalog within transaciton - this trick is used by other tests too.